10_datastructures_lab/01.py

```diff
@@ -1,10 +1,83 @@
-"""
-Write a program that takes two strings
-from the user and checks if they represent
-a valid user name.
-Valid users and passwords:
-    apple => red
-    lettuce => green
-    lemon => yellow
-    orange => orange
-"""
+#!/usr/bin/env python
+
+"""
+This program parse the given hosts file and return the IP address of a given computers list
+
+"""
+__author__ = 'Eabel'
+
+import sys
+import os
+import argparse
+import socket
+from collections import defaultdict
+
+
+def is_valid_ipv4_address(address):
+    try:
+        socket.inet_pton(socket.AF_INET, address)
+    except AttributeError:  # no inet_pton here, sorry
+        try:
+            socket.inet_aton(address)
+        except socket.error:
+            return False
+        return address.count('.') == 3
+    except socket.error:  # not a valid address
+        return False
+
+    return True
+
+def is_valid_ipv6_address(address):
+    try:
+        socket.inet_pton(socket.AF_INET6, address)
+    except socket.error:  # not a valid address
+        return False
+    return True
+
+def is_valid_ip_address(address):
+    return is_valid_ipv4_address or is_valid_ipv6_address(address)
+
+
+def read_hosts_file(file_path):
+    try:
+        f = open(file_path)
+        host_dict = defaultdict(str)
+        for line in f:
+            items = line.split('=')
+            if len(items) != 2:
+                return False;
+            if not is_valid_ip_address(items[1]):
+                return False
+            host_dict[items[0]] = items[1].rstrip('\r\n')
+        return host_dict
+    except:
+        e = sys.exc_info()[0]
+        print "Error loading hosts file -",e
+        return False
+
+def get_ip(machine,host_dict):
+    ip = host_dict[machine];
+    if not ip:
+        print 'Error -',machine,'is missing from hosts file'

```

Comment:
Nice!

Sent from my iPhone

> On 10 Jul 2016, at 22:20, ynonp <notifications@github.com> wrote:
> 
> In 10_datastructures_lab/01.py:
> 
> > +            items = line.split('=')
> > +            if len(items) != 2:
> > +                return False;
> > +            if not is_valid_ip_address(items[1]):
> > +                return False
> > +            host_dict[items[0]] = items[1].rstrip('\r\n')
> > +        return host_dict
> > +    except:
> > +        e = sys.exc_info()[0]
> > +        print "Error loading hosts file -",e
> > +        return False
> > +
> > +def get_ip(machine,host_dict):
> > +    ip = host_dict[machine];
> > +    if not ip:
> > +        print 'Error -',machine,'is missing from hosts file'
> Better to write the error message inside sys.exit(). It prints the error to stderr AND returns an error code to the operating system.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>